src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj

```diff
@@ -170,9 +170,6 @@
       <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)**\NuGet*.resources.dll">
         <PackagePath>CoreCLR/</PackagePath>
       </TfmSpecificPackageFile>
-      <TfmSpecificPackageFile Include="$(OutputPath)\$(ILMergeSubpath)**\*.dll" Condition="'$(IsBuildOnlyXPLATProjects)' == 'true'">
```

Comment:
Why are we removing this? 

The one difference between source build and non source build sdks was that non source build ilmerged.

Reply:
It is adding additional NuGet assemblies that differs from the final sdk shipped by Microsoft. The difference was noted in this [comment](https://github.com/dotnet/source-build/issues/708#issuecomment-416818823). Tracking issue in source-build [here](https://github.com/dotnet/source-build/issues/739).

Reply:
In addition, there's a NU5118 warning in when NuGet is compiled via source-build (edited) when this patch is not applied:

```
/usr/local/share/dotnet/sdk/5.0.100-preview.4.20258.7/Sdks/NuGet.Build.Tasks.Pack/buildCrossTargeting/NuGet.Build.Tasks.Pack.targets(198,5): warning NU5118: File '$REPO_ROOT/artifacts/NuGet.Build.Tasks.Pack/16.0/bin/Debug/netstandard2.0/NuGet.Build.Tasks.Pack.dll' is not added because the package already contains file 'CoreCLR/NuGet.Build.Tasks.Pack.dll' [$REPO_ROOT/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj]
```

Reply:
Ok, my comment here is that at the time the current behavior was a deliberate decision.